packages/contracts/test/exchange/core.ts

```diff
@@ -469,237 +475,135 @@ describe('Exchange', () => {
-            expect(taker).to.be.equal(logArgs.taker);
-            expect(signedOrder.feeRecipient).to.be.equal(logArgs.feeRecipient);
-            expect(signedOrder.makerTokenAddress).to.be.equal(logArgs.makerToken);
-            expect(signedOrder.takerTokenAddress).to.be.equal(logArgs.takerToken);
+            expect(signedOrder.makerAddress).to.be.equal(logArgs.makerAddress);
+            expect(takerAddress).to.be.equal(logArgs.takerAddress);
+            expect(signedOrder.feeRecipientAddress).to.be.equal(logArgs.feeRecipientAddress);
+            expect(signedOrder.makerTokenAddress).to.be.equal(logArgs.makerTokenAddress);
+            expect(signedOrder.takerTokenAddress).to.be.equal(logArgs.takerTokenAddress);
             expect(expectedFilledMakerTokenAmount).to.be.bignumber.equal(logArgs.makerTokenFilledAmount);
             expect(expectedFilledTakerTokenAmount).to.be.bignumber.equal(logArgs.takerTokenFilledAmount);
-            expect(expectedFeeMPaid).to.be.bignumber.equal(logArgs.makerFeePaid);
-            expect(expectedFeeTPaid).to.be.bignumber.equal(logArgs.takerFeePaid);
-            expect(ZeroEx.getOrderHashHex(signedOrder)).to.be.equal(logArgs.orderHash);
+            expect(expectedFeeMPaid).to.be.bignumber.equal(logArgs.makerFeeAmountPaid);
+            expect(expectedFeeTPaid).to.be.bignumber.equal(logArgs.takerFeeAmountPaid);
+            expect(orderUtils.getOrderHashHex(signedOrder)).to.be.equal(logArgs.orderHash);
         });
 
         it('should throw when taker is specified and order is claimed by other', async () => {
-            signedOrder = await orderFactory.newSignedOrderAsync({
-                taker: feeRecipient,
+            signedOrder = orderFactory.newSignedOrder({
+                takerAddress: feeRecipientAddress,
                 makerTokenAmount: ZeroEx.toBaseUnitAmount(new BigNumber(100), 18),
                 takerTokenAmount: ZeroEx.toBaseUnitAmount(new BigNumber(200), 18),
             });
-
-            return expect(exWrapper.fillOrderAsync(signedOrder, taker)).to.be.rejectedWith(constants.REVERT);
+            return expect(exWrapper.fillOrderAsync(signedOrder, takerAddress)).to.be.rejectedWith(constants.REVERT);
         });
 
         it('should throw if signature is invalid', async () => {
-            signedOrder = await orderFactory.newSignedOrderAsync({
+            signedOrder = orderFactory.newSignedOrder({
                 makerTokenAmount: ZeroEx.toBaseUnitAmount(new BigNumber(10), 18),
             });
 
-            signedOrder.ecSignature.r = ethUtil.bufferToHex(ethUtil.sha3('invalidR'));
-            signedOrder.ecSignature.s = ethUtil.bufferToHex(ethUtil.sha3('invalidS'));
-            return expect(exWrapper.fillOrderAsync(signedOrder, taker)).to.be.rejectedWith(constants.REVERT);
+            const invalidR = ethUtil.sha3('invalidR');
+            const invalidS = ethUtil.sha3('invalidS');
+            const invalidSigBuff = Buffer.concat([
```

Comment:
Should this have the SignatureType concatenated with it?

Reply:
Yep, that's why we're slicing the 6 characters (2 bytes) rather than 4 (1 byte).